### PR TITLE
Display auth login modal

### PR DESCRIPTION
**Pull request by elayabharath**
+ Changed share buttons to logout button
+ Removed default loading message to move it the app later

image

### CODE REVIEW

File: lib/firebase_constants.dart [hunk 1]
```diff
@@ -27,3 +27,5 @@ String get themesDoc => _constants['themesDoc'];
 String get dataDoc => _constants['dataDoc'];
 String get peopleCollection => _constants['peopleCollection'];
 String get messagesCollection => _constants['messagesCollection'];
+List<String> get allowedEmailDomains =>
```
Comment (lukechurch) [inline lib/firebase_constants.dart:31]: I don't really follow your intention here - this configuration needs to be done on the server not the client - let me know if that doesn't make sense and you need a hand with how we setup this kind of thing

File: web/index.html [hunk 2]
```diff
@@ -63,5 +60,15 @@ <h1>Title of the chart</h1>
       </div>
     </div>
   </footer>
+
+  <!-- Login modal -->
+  <div class="modal-wrapper" id="login-modal">
+    <div class="modal-content">
+      <h5>This demo is meant to be private to Africa's voices.</h5>
```
Comment (lukechurch) [inline web/index.html]: Let's make this a bit more intentional.

"This demo is only accessible to Africa's Voices. Please reach out if you'd like to see a demo"

File: web/index.html [hunk 3]
```diff
@@ -63,5 +60,15 @@ <h1>Title of the chart</h1>
...
+      <p>Please reach out if you like access.</p>
+      <button class="btn btn-danger" id="login-button">Login via Google SSO</button>
```
Comment (lukechurch) [inline web/index.html]: 'Login with Google' - is the way that OAuth SSO is usually described to users

**Review by lukechurch — CHANGES_REQUESTED** [review]
I need to understand the email filtering setup you're using a bit more before we merge this. Thanks @elayabharath

File: lib/firebase_constants.dart (hunk 1 above)
Comment (elayabharath) [reply lib/firebase_constants.dart:31]: The actual access control will be done on the server. This is purely to show an error message in the client that says along the lines of "the email under this domain is not permitted". So when the user signs in with say @gmail.com - I can catch the email to see if it ends with the domains in the config and show the notification in the client.

Is this the way to do this or is there any other way to whitelist allowed email domains for authentication? @lukechurch

File: web/index.html (hunk 2 above)
Comment (elayabharath) [reply web/index.html]: Sure, noted

File: web/index.html (hunk 3 above)
Comment (elayabharath) [reply web/index.html]: All the examples in official google docs say "Sign in with Google" - will change it to the recommended one.

File: lib/firebase_constants.dart (hunk 1 above)
Comment (marianamarasoiu) [reply lib/firebase_constants.dart:31]: In reviewing the other PR I ended up looking into this as well. I didn't know this until now, but it turns out that the way Firebase and Google OAuth integrate, any user can "create an account" by logging in by default, and you need to have multiple checks to support the kind of restricted logging in we're trying to do.

[This stackoverflow answer](https://stackoverflow.com/a/44347561) summarises what needs to be done: we need the firebase rules to specify read/writes on the database, but we also need to handle users that go through the login process but who are not allowed to read/write (ie. the domain filtering), but who have been logged in. In Coda we show that there are no datasets available by having an empty list, or show the missing permissions error when trying to access a project that the user doesn't have permissions to access, while Nook doesn't really handle this other than throwing errors in the console.

So from what I can tell, checking that the email of the logged in user is part of an allowed domain in the client like this seems to be the recommended solution, even if it's kinda ugly.

@elayabharath May I suggest adding a comment here that `allowedEmailDomains` should match the list of allowed domains in the firebase rules file?

File: web/index.html (hunk 3 above)
Comment (elayabharath) [reply web/index.html]: Fixed

File: web/index.html (hunk 2 above)
Comment (elayabharath) [reply web/index.html]: Fixed

File: lib/firebase_constants.dart (hunk 1 above)
Comment (elayabharath) [reply lib/firebase_constants.dart:31]: Thanks for the summary. Added the comment in the place where we read the domains since json doesn't have comments.

**Review by lukechurch — APPROVED** [review]
LGTM - along with a server side test.

Thanks @elayabharath, one quick request, after making changes like this could you click 'request a review again' on me so that it re-enters my standard todo list for reviewing? Thanks!

**Comment by elayabharath** [comment]
Thanks @lukechurch Noted about the request a review again

**Pull request merged**